lang/scala/scala.py

```diff
@@ -8,32 +8,81 @@
 
 # Scala Common Types
 scala_common_types = {
-    "Byte": "Byte",
-    "Int": "Int",
-    "Double": "Double",
-    "Short": "Short",
-    "Float": "Float",
-    "Long": "Long",
-    "Boolean": "Boolean",
-    "Char": "Char",
-    "String": "String",
-    "None": "None"
+    "boolean": "Boolean",
+    "int": "Int",
+    "float": "Float",
+    "byte": "Byte",
+    "double": "Double",
+    "long": "Long",
+    "char": "Char",
+    "unit": "Unit",
+    "any": "Any",
+    "any val": "AnyVal",
+    "string": "String",
+    "thread": "Thread",
+    "exception": "Exception",
+    "throwable": "Throwable",
+    "none": "None",
+    "success": "Success",
+    "failure": "Failure",
 }
 
 # Scala Common Generic Types
 scala_common_generic_types = {
-    "seq": "Seq",
-    "set": "Set",
-    "list": "List",
     "array": "Array",
+    "deque": "Deque",
+    "future": "Future",
+    "list": "List",
     "map": "Map",
-    "stack": "Stack"
+    "nil": "Nil",
+    "option": "Option",
+    "queue": "Queue",
+    "seq": "Seq",
+    "set": "Set",
+    "some": "Some",
+    "stack": "Stack",
+    "try": "Try",
 }
 
 scala_types = scala_common_types.copy()
 scala_types.update(scala_common_generic_types)
 ctx.lists["user.code_type"] = scala_types
 
+# Scala Modifies
+scala_modifiers = {
+    "public": "public",
+    "private": "private",
+    "protected": "protected",
+}
+
+mod.list("scala_modifier", desc="Scala Modifiers")
+ctx.lists["user.scala_modifier"] = scala_modifiers
+
+scala_keywords = {
+    "abstract": "abstract ",
+    "case class": "case class ",
+    "class": "class ",
+    "def": "def ",
```

Comment:
so `def` is how you define a function and so is already covered by "funk <blah>", as is I think `package` and possibly others below. @splondike not sure about default repo's philosophy regarding duplicate ways of doing things, but thought I'd flag if you weren't aware.

Reply:
This is mostly true, but at least in our codebase, we also sometimes use `def` for expressions without argument brackets:
```scala
def foo = "hello"
// or
override def bar = "baz"
```
afaik, `funky` inserts `()` automatically. So I'm not sure this counts as duplicate functionality. What do you think?

Reply:
Yep that's fair. Scala is nothing if not flexible in how you use it